frame/balances/src/lib.rs

```diff
@@ -1682,8 +1689,8 @@ where
 		who: &T::AccountId,
 		amount: T::Balance,
 		reasons: WithdrawReasons,
-	) {
-		if amount.is_zero() || reasons.is_empty() { return }
+	) -> DispatchResult {
+		if amount.is_zero() || reasons.is_empty() { return Ok(()) }
 		let mut new_lock = Some(BalanceLock { id, amount, reasons: reasons.into() });
 		let mut locks = Self::locks(who).into_iter().filter_map(|l|
```

Comment:
Same pattern applies here. We don't need to reallocate the vector twice; we can just mutate it.